### ISSUE TITLE
Problem with STDP Learning

### ISSUE THREAD

**Issue by youyou782** [fangwei123456/spikingjelly#301]
Hi!
I'm doing some experiment on the STDP Learning using spikejelly following the tutorials. Genellary, the tutorial only show one iteration  of STDP Learning. When I try to repeat the training process (that is, repeat the following codes and append them to the original code), some error occurs.

`x_seq = (torch.rand([T, N, C, H, W]) > 0.5).float()
target = torch.randint(low=0, high=10, size=[N])

optimizer_gd.zero_grad()
optimizer_stdp.zero_grad()

y = net(x_seq).mean(0)
loss = F.cross_entropy(y, target)
loss.backward()

optimizer_stdp.zero_grad()

for i in range(stdp_learners.__len__()):
    stdp_learners[i].step(on_grad=True)

optimizer_gd.step()
optimizer_stdp.step()`

It seems like when running `loss.backward()` for the second time, the computing graph has already been freed. I think it should not happen, because the feedforward process has already happened which should have generated a new computing graph.

Error explanation
`发生异常: RuntimeError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
Trying to backward through the graph a second time (or directly access saved tensors after they have already been freed). Saved intermediate values of the graph are freed when you call .backward() or autograd.grad(). Specify retain_graph=True if you need to backward through the graph a second time or if you need to access saved tensors after calling backward.`

**Comment by fangwei123456** [comment]
https://spikingjelly.readthedocs.io/zh_CN/latest/activation_based_en/basic_concept.html#saving-and-resetting-of-states

![image](https://user-images.githubusercontent.com/38352109/200546911-b9f0c08d-463c-4cf2-a712-2d631b4349be.png)

**Comment by youyou782** [comment]
Oh, thanks so much for your generous help. It works well now.
